### PR TITLE
Feat play tie game

### PR DESCRIPTION
**Pull request by parkerpersaud**

### CODE REVIEW

File: src/main/java/me/karakelley/tictactoe/BoardState.java [hunk 1]
```diff
@@ -1,12 +1,16 @@
 package me.karakelley.tictactoe;
 
+import javax.swing.text.html.StyleSheet;
```
Comment (aaronlahey) [inline src/main/java/me/karakelley/tictactoe/BoardState.java]: 🤔

File: src/main/java/me/karakelley/tictactoe/BoardState.java [hunk 2]
```diff
@@ -19,7 +23,96 @@ public BoardState() {
     return grid;
   }
 
+  public boolean fullBoard() {
```
Comment (aaronlahey) [inline src/main/java/me/karakelley/tictactoe/BoardState.java:21]: No need to refactor this, but you can implement this without a temp variable...

```
for (int i = 0; i < gridLength(); i++)
  if (cellBlank(i))
    return false;
return true;
```

**Review by aaronlahey — APPROVED** [review]
I would run a code cleanup before merging this. Also, there seems to be a lot of refactoring that isn't related to playing a full game... It is typically better to split refactoring and feature work into separate PRs, even if the refactor is just a "prefactor" to the actual feature.  Typically I'll put up a prefactor PR with an explanation of why I'm doing it and how it will make the upcoming feature easier.

File: src/main/java/me/karakelley/tictactoe/BoardState.java [hunk 3]
```diff
@@ -19,7 +23,96 @@ public BoardState() {
...
+  private boolean cellBlank(int cell) {
+    return grid[cell] == "-";
+  }
+
+  private int gridLength() {
+    Double gridLength = Math.pow(boardSize, 2);
+    return gridLength.intValue();
+  }
+
+  private String[][] rows() {
```
Comment (andarcabrera) [inline src/main/java/me/karakelley/tictactoe/BoardState.java:65]: Nice job making these work for any size board

File: src/main/java/me/karakelley/tictactoe/BoardState.java [hunk 4]
```diff
@@ -19,7 +23,96 @@ public BoardState() {
...
+      }
+    }
+    return full;
+  }
+
+  public boolean cellAvailable(int cell) {
+    return cellInBounds(cell) && cellBlank(cell);
+  }
+
+  public String[][] combinations() {
```
Comment (andarcabrera) [inline src/main/java/me/karakelley/tictactoe/BoardState.java]: I'd name them `winningCombinations` for more clarity

File: src/main/java/me/karakelley/tictactoe/Game.java [hunk 5]
```diff
@@ -1,30 +1,35 @@
...
+    return winner;
   }
 
-  private void play() {
-    String chosenCell = userInterface.userPrompt("Please enter an available cell:\n", validator.make("NumberValidator"));
-    boardState.placeMove(chosenCell, humanPlayer.getMarker());
-    userInterface.displayBoard(boardState.getGrid());
+  private List<String> removeDuplicates(String[] combinations){
+    List<String> combos = Arrays.asList(combinations);
+    return combos.stream().distinct().collect(Collectors.toList());
```
Comment (andarcabrera) [inline src/main/java/me/karakelley/tictactoe/Game.java]: This is just a stylistic preference, but I find it more readable when you use a new line for each method, when you chain like this:

e.g. 

``` 
    combos.stream()
          .distinct()
          .collect()
          .etc
```

File: src/main/java/me/karakelley/tictactoe/GameLoop.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    while (!game.gameOver(boardState)) {
+      nextPlayersTurn(boardState, game);
+    }
+    if (game.tie(boardState)) {
+      userInterface.displayMessage("It's a tie!\n");
+    }
+  }
+
+  private void nextPlayersTurn(BoardState boardState, Game game) {
+    String chosenCell = userInterface.userPrompt("Please enter an available cell:\n", new NumberValidator(boardState));
```
Comment (andarcabrera) [inline src/main/java/me/karakelley/tictactoe/GameLoop.java:38]: Many of the messages are spread around. What will have to change if your game needs to support 2 languages?

**Review by andarcabrera — APPROVED** [review]

**Pull request merged**